data/src/main/scala/io/prediction/data/api/EventServer.scala

```diff
@@ -96,18 +100,34 @@ class EventServiceActor(
             }.getOrElse{
               Right(AuthData(k.appid, None))
             }
-          }.getOrElse{
-            Left(AuthenticationFailedRejection(
-              AuthenticationFailedRejection.CredentialsRejected, List()))
-          }
-        }.getOrElse { Left(AuthenticationFailedRejection(
-          AuthenticationFailedRejection.CredentialsMissing, List()))
+          }.getOrElse(FailedAuth)
+        }.getOrElse {
+          // with accessKey in header, return appId if succeed
```

Comment:
is it because SegmentIO does not support specifying accessKey as query parameter?